src/components/Table/components/cell.tsx

```diff
@@ -34,6 +34,8 @@ export type GridCellContent = {
   revealOnRowHover?: true;
   /** Tooltip to add to a cell */
   tooltip?: ReactNode;
+  /** Allows cell to be editable when hovering, and also highlights the field on hover */
+  editableOnHover?: true;
```

Comment:
@0ces cc @apattersonATX-HB do we want this to be a per-cell flag? Kinda thinking that the entire table should be consistent about "all columns/cells are `editableOnHover` yes/no"...

Reply:
Setting at the table makes sense. Maybe have an override on the cell/column level?